source/app.d

```diff
@@ -220,7 +235,20 @@ LRESULT WndProc(HWND hwnd, uint msg, WPARAM wParam, LPARAM lParam) nothrow {
             SendMessage(hwnd, WM_CLOSE, 0, 0);
             break;
 
-            default: break;
+            default:
+            uint newLayoutIdx = cast(uint)wParam - ID_LAYOUTMENU;
+            // Did the user select a valid layout index?
+            if (newLayoutIdx >= 0 && newLayoutIdx < layouts.length) {
+                configStandaloneLayout = &layouts[newLayoutIdx];
+                checkKeyboardLayout();
+                updateTrayTooltip();
+                CheckMenuRadioItem(layoutMenu, 0, GetMenuItemCount(layoutMenu) - 1, newLayoutIdx, MF_BYPOSITION);
+                // Persist new selected layout
+                auto configJson = parseJSONFile("config.json");
```

Comment:
Ja, das abspeichern hier so mittendrin zu machen ist langfristig nicht optimal. Für jetzt™ passt es aber erstmal.

Reply:
Nicht nur das, auch die Tatsache, dass man das Event so im Default-Case-Zweig abhandelt, gefällt mir nicht so richtig. Das ganze Tray-Menü-Zeug würde ich später lieber in eine kleine Klasse stecken, wie bei dem Trayicon, damit die `app.d` klein bleibt. (Neben anderen Wünschen fürs Refactoring, hab da schon eine Liste.)